### ISSUE TITLE
Feature Request:  Could a Train/Validation/Test option be added as an extension to vfold_cv.  Maybe vfold_tcv?

### ISSUE THREAD

**Issue by UnclAlDeveloper** [tidymodels/rsample#169]
Could a vfold function be added that includes a Test fold as well as a Validation fold.  So for example if you choose 10 folds, then 8 would be for training, one would be for validation (used to tune the model), and one would be completely out-of-sample.  The reason for this is that the out-of-sample (Test) predictions should be returned in *tune::collect_predictions*.  This would allow all data to be used and give an accurate out-of-sample representation of how well a model worked.  Tuning parameters are mostly designed to stop over-fitting, but they can also be guilty of slightly fitting in *tune*.  ie.  Picking the best hyper-parameter set out of a hundred possibilities, isn't over-fitting, but it is fitting, and means that the predictions are not truly out-of-sample.  Having a Test/CV vfold function would mean that you can produce fully out-of-sample predictions over the full dataset.

**Comment by UnclAlDeveloper** [comment]
Obviously, repeats would be removed from the parameters, since repeats could only be 1.  But a vfold_tcv function would be fulfilling a different required niche than vfold_cv.

**Comment by UnclAlDeveloper** [comment]
Actually I'm not sure repeats would need to be removed since you could just average over the OOS (Test) folds if repeats were greater than 1.

**Comment by topepo** [comment]
We support binary splits in resampling. There is `validation_split()` that would be used to create a validation set (once).
